salt/docker/init.sls

```diff
@@ -16,6 +16,7 @@ include:
         Environment="NO_PROXY={{ salt['pillar.get']('proxy:no_proxy', '') }}"
   cmd.run:
     - name: systemctl daemon-reload
+    - stateful: True
```

Comment:
I can confirm this worked:

```
/tmp/somefile:
  file.managed:
    - source: /etc/systemd/system/docker.service.d/proxy.conf
    - watch:
      - file: /etc/systemd/system/docker.service.d/proxy.conf
```

When `/etc/systemd/system/docker.service.d/proxy.conf` changed its contents, `/tmp/somefile` did too, even with the `cmd.run` with `stateful: True`.